api-controllers/DashboardController.js

```diff
@@ -136,6 +141,50 @@ class DashboardController extends FabrikBaseController {
       .throw(new ContinueWithNext());
   }
 
+  validateServiceInstanceAndType(req, res) {
+    /* jshint unused:false */
+    const instance_id = req.params.instance_id;
+    const instance_type = req.params.instance_type;
+    return this._getApiServerResource(instance_id, instance_type)
+      .then(resource => _.get(resource, 'spec.options'))
+      .then(resourceOptions => {
+        const service_id = _.get(resourceOptions, 'service_id');
+        const plan_id = _.get(resourceOptions, 'plan_id');
+        const context = _.get(resourceOptions, 'context');
+        req.params.service_id = service_id;
+        req.params.plan_id = plan_id;
+        return this.fabrik.createInstance(instance_id, service_id, plan_id, context);
+      })
+      .then(instance => {
+        req.instance = instance;
+        req.manager = instance.manager;
+      })
+      .throw(new ContinueWithNext());
+  }
+
+  _getApiServerResource(instance_id, instance_type) {
+    switch (instance_type) {
+    case CONST.INSTANCE_TYPE.DIRECTOR:
```

Comment:
since most of the code is same , ideally you could have just assigned resourceType to a variable and get the resource only once outside the switch ?

Reply:
done